core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -233,7 +233,8 @@ class ReplicaFetcherThread(name: String,
         Map.empty
       }
     } else {
-      fetchResponse.responseData(fetchSessionHandler.sessionTopicNames, clientResponse.requestHeader().apiVersion()).asScala
+      fetchResponse.responseData(fetchSessionHandler.sessionTopicNames, clientResponse.requestHeader().apiVersion()).asScala.map {
```

Comment:
Not related to this line. Don't wee need to update the fetcher to handle the topic id errors at the partition level?